genesyscloud/data_source_genesyscloud_auth_division.go

```diff
@@ -45,8 +45,13 @@ func dataSourceAuthDivisionRead(ctx context.Context, d *schema.ResourceData, m i
 			return retry.RetryableError(fmt.Errorf("No authorization divisions found with name %s", name))
 		}
 
-		division := (*divisions.Entities)[0]
-		d.SetId(*division.Id)
-		return nil
+		for _, division := range *divisions.Entities {
```

Comment:
Excellent.  I was just looking through the code on for this resource.  Do we want to refactor this resource while we are at it. @dginty4 Thoughts